### ISSUE TITLE
Remove promise wrapper

### ISSUE THREAD

**Issue by fat4lix** [tserkov/vue-sse#7]
Hello! I think there is no need return Promise from $sse factory function. Return a promise causes some problem.  If sse server not available when component mounted, reject function triggered and resolve will never be fired.
This will lead to the fact that we will not get the wrapper object when restoring the connection to the server. This way we lose the parent EventSource object in the closure and can't close it. This situation is especially dangerous in SPA applications. I believe that EventSource already provides the necessary functions for working with async and Promise is a redundant solution here

**Comment by tserkov** [comment]
> If sse server not available when component mounted, reject function triggered and resolve will never be fired.
That is exactly as intended.  If the connection fails, the promise is rejected, and it is up to you, the caller, to implement the retry functionality.

> This will lead to the fact that we will not get the wrapper object when restoring the connection to the server.
You only get the wrapper object on successful connection. Just keep calling `$sse` until you connect, and you will have the wrapper object.

> This way we lose the parent EventSource object in the closure and can't close it.
If you never connect, there is nothing to close.

**Comment by fat4lix** [comment]
I'm sorry, but you're wrong.

EventSource is designed in such a way that after creating an object (**new EventSource**), it independently makes connection attempts until the connection is established, and when the connection is lost (**onerror**), it also tries to restore the connection itself. To interrupt the connection process and destroy the object we need to explicitly call the **close()** function
Try creating several objects with an unavailable address in the console and you will see how they will both try to connect one after the other, hanging in memory forever. In your case, you create an **EventSource** object in a loop and give direct access to it only after it has established a connection, but you will never get this access if the server was unavailable when creating the **EventSource** object, because the reject handler is hung on the **onerror** event. 

I will describe the problem scenario when the server is unavailable

You call $ssr funcitons
It creates an EventSource object in its scope
When creating an EventSource, it tries to make an attempt to connect and gets an error. onerror event on EventSource object is triggered and therefore reject in your Promise. In this case, the EventSource remains in memory forever and will make endless attempts to restore the connection. And even if the connection is restored, you will never get a wrapper object to close this connection, because resolve will not be called anymore (reject has already been called)

**Comment by tserkov** [comment]
https://html.spec.whatwg.org/multipage/server-sent-events.html#sse-processing-model
> Once the user agent has failed the connection, it does not attempt to reconnect!

**Comment by fat4lix** [comment]
Ok, so what it is ?
https://monosnap.com/file/wGAMQrh3PkLIRTarJD7o0W3wWVzpYn

**Comment by sbernard31** [comment]
@fat4lix how do you deal with this issue finally ? 
(I face another issue related to returning a promise :  #12)

**Comment by tserkov** [comment]
One year later, with the addition of @sbernard31's issue, I realize the critical design flaw in this library.

By only returning the VueSSE wrapper on successful connections, consumers have no way of closing the connection if it connects after the calling component has gone out of scope/destroyed.

I will release a v2 of this library that provides a mechanism for better controlling the lifecycle in the near future.

My apologies to @fat4lix for my delayed comprehension.
